src/protocols/screencopy.rs

```diff
@@ -0,0 +1,279 @@
+        _client: &Client,
+        manager: &ZwlrScreencopyManagerV1,
+        request: zwlr_screencopy_manager_v1::Request,
+        _data: &(),
+        _display: &DisplayHandle,
+        data_init: &mut DataInit<'_, D>,
+    ) {
+        let (frame, overlay_cursor, rect, output) = match request {
+            zwlr_screencopy_manager_v1::Request::CaptureOutput {
+                frame,
+                overlay_cursor,
+                output,
+            } => {
+                let output = Output::from_resource(&output).unwrap();
+                let rect =
+                    Rectangle::from_loc_and_size((0, 0), output.current_mode().unwrap().size);
+                (frame, overlay_cursor, rect, output)
+            }
+            zwlr_screencopy_manager_v1::Request::CaptureOutputRegion {
+                frame,
+                overlay_cursor,
+                x,
+                y,
+                width,
+                height,
+                output,
+            } => {
+                let output = Output::from_resource(&output).unwrap();
+                let rect = Rectangle::from_loc_and_size((x, y), (width, height));
+
+                // Translate logical rect to physical framebuffer coordinates.
+                let output_transform = output.current_transform();
+                // let rotated_rect = output_transform.transform_rect_in(rect,
+                // &output.current_mode().unwrap().size.to_f64().to_logical(output.current_scale()).
+                // to_i32_round()); let physical_rect =
+                // rotated_rect.to_physical_precise_round(output.current_scale().integer_scale());
+                let physical_rect =
+                    output_transform.transform_rect_in(rect, &output.current_mode().unwrap().size);
+
+                // Clamp captured region to the output.
+                let clamped_rect = physical_rect
+                    .intersection(Rectangle::from_loc_and_size(
+                        (0, 0),
+                        output.current_mode().unwrap().size,
+                    ))
+                    .unwrap_or_default();
+
+                (frame, overlay_cursor, clamped_rect, output)
+            }
+            zwlr_screencopy_manager_v1::Request::Destroy => return,
+            _ => unreachable!(),
+        };
+
+        // Create the frame.
+        let overlay_cursor = overlay_cursor != 0;
+        let frame = data_init.init(
+            frame,
+            ScreencopyFrameState {
+                output,
+                overlay_cursor,
+                rect,
+            },
+        );
+
+        // Send desired SHM buffer parameters.
+        frame.buffer(
+            wl_shm::Format::Argb8888,
+            rect.size.w as u32,
+            rect.size.h as u32,
+            rect.size.w as u32 * 4,
+        );
+
+        if manager.version() >= 3 {
+            // Send desired DMA buffer parameters.
+            frame.linux_dmabuf(
+                Fourcc::Argb8888 as u32,
```

Comment:
I think the "default format" is actually `Abgr8888`, that's what we use for dmabufs in screenshot etc., let's use it here too.